x-pack/solutions/observability/plugins/observability_agent_builder/server/routes/ai_insights/route.ts

```diff
@@ -30,7 +31,95 @@ export function getObservabilityAgentBuilderAiInsightsRouteRepository() {
     },
   });
 
+  const logAiInsightsRoute = createObservabilityAgentBuilderServerRoute({
+    endpoint: 'POST /internal/observability_agent_builder/ai_insights/log',
+    options: {
+      access: 'internal',
+    },
+    security: {
+      authz: {
+        requiredPrivileges: [apiPrivileges.readOnechat],
+      },
+    },
+    params: t.type({
+      body: t.type({
+        fields: t.record(t.string, t.any),
+        connectorId: t.union([t.string, t.undefined]),
+      }),
+    }),
+    handler: async (args) => {
+      const { request, core, plugins, dataRegistry, params, logger } = args;
+      const { fields, connectorId: lastUsedConnectorId } = params.body;
+
+      const [_, pluginsStart] = await core.getStartServices();
+
+      let connectorId = lastUsedConnectorId;
+      if (!connectorId) {
+        const defaultConnector = await pluginsStart.inference.getDefaultConnector(request);
+        connectorId = defaultConnector?.connectorId;
+      }
+
+      if (!connectorId) {
+        throw new Error('No default connector found');
+      }
+
+      const inferenceClient = pluginsStart.inference.getClient({ request });
+
+      const [coreStart] = await core.getStartServices();
+      const esClient = coreStart.elasticsearch.client.asScoped(request);
+      const index = fields.find((field) => field.field === '_index')?.value;
+      const id = fields.find((field) => field.field === '_id')?.value;
+      const result = await esClient.asCurrentUser.search({
+        index,
+        query: {
+          bool: {
+            filter: [
+              {
+                ids: {
+                  values: id,
+                },
+              },
+            ],
+          },
+        },
+        fields: [
+          {
+            field: '*',
+            include_unmapped: true,
+          },
+        ],
+      });
+      const logEntry = result.hits.hits[0]._source;
+      const logEntryFields = result.hits.hits[0].fields;
+      if (!logEntry) {
+        throw new Error('Log entry not found');
+      }
+
+      const TWENTY_FOUR_HOURS_MS = 24 * 60 * 60 * 1000;
+      const logTimestamp = new Date(logEntry['@timestamp']).getTime();
+      const serviceSummary = await dataRegistry.getData('apmServiceSummary', {
+        request,
+        serviceName: logEntry.service.name,
+        serviceEnvironment: logEntry.service.environment,
+        start: new Date(logTimestamp - TWENTY_FOUR_HOURS_MS).toISOString(),
+        end: new Date(logTimestamp + TWENTY_FOUR_HOURS_MS).toISOString(),
```

Comment:
Use moment for date calculations (yes, moment is bloated and deprecated but it's the lib we have)